api/src/main/java/org/apache/iceberg/PartitionField.java

```diff
@@ -67,8 +67,7 @@ public String toString() {
   public boolean equals(Object other) {
     if (this == other) {
       return true;
-    }
-    if (other == null || getClass() != other.getClass()) {
+    } else if (!(other instanceof PartitionField)) {
```

Comment:
nit: I feel multiple `if (...) { return ... }` is more readable than `if ... else ...`.

Reply:
Normally I would agree with you, but in both statements, there is a return. So it isn't possible to enter both `if` statements and `else if` makes more sense to me then.